public void testReferenceChain() throws IOException {
    ObjectMapper mapper = new ObjectMapper();
    JsonMappingException exception = expectThrows(JsonMappingException.class, () ->
        mapper.readValue(mapper.writeValueAsBytes(new Outer()), Outer.class));
    assertEquals("ReferenceChainTest$Outer[\"inner\"]", exception.getPath().get(0).toString());
}

static class Outer {
    public Inner inner = new Inner();
}

static class Inner {
    public int x;

    @JsonCreator public static Inner create(@JsonProperty("x") int x) {
        throw new RuntimeException("test-exception");
    }
}
